### PR TITLE
Implemented stack

### LINKED ISSUE #9: Stack
Implement a complete and functioning stack

### PR DESCRIPTION
**Pull request by 1c3t3a**
Closes #9.

### CODE REVIEW

**Comment by 1c3t3a** [comment]
# [Codecov](https://codecov.io/gh/1c3t3a/rust-data-structures/pull/20?src=pr&el=h1) Report
> :exclamation: No coverage uploaded for pull request base (`master@3fe702e`). [Click here to learn what that means](https://docs.codecov.io/docs/error-reference#section-missing-base-commit).
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/1c3t3a/rust-data-structures/pull/20?src=pr&el=tree)

```diff
@@            Coverage Diff            @@
##             master      #20   +/-   ##
=========================================
  Coverage          ?   96.28%           
=========================================
  Files             ?        2           
  Lines             ?      377           
  Branches          ?        0           
=========================================
  Hits              ?      363           
  Misses            ?       14           
  Partials          ?        0           
```



------

[Continue to review full report at Codecov](https://codecov.io/gh/1c3t3a/rust-data-structures/pull/20?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/1c3t3a/rust-data-structures/pull/20?src=pr&el=footer). Last update [3fe702e...bd4610d](https://codecov.io/gh/1c3t3a/rust-data-structures/pull/20?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

File: src/stack.rs [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+    /// Creates a new empty stack.
+    /// ```rust
+    /// use data_structure_with_colin::stack::Stack;
+    /// let mut stack = Stack::<()>::new();
+    /// assert!(stack.is_empty());
+    /// ```
+    pub fn new() -> Self {
+        Stack { first: None }
+    }
+
```
Comment (Colin1860) [inline src/stack.rs:45]: Suggestion: Think about implementing the Default trait

File: src/stack.rs [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+        Iter { head: &self.first }
+    }
+}
+
+impl<T: Eq> Node<T> {
+    fn new(value: T) -> Self {
+        Node { next: None, value }
+    }
+
+    fn size(&self) -> i32 {
```
Comment (Colin1860) [inline src/stack.rs:140]: I'm not happy with a non storing size function here tbh, could you store the size in a variable pls

Comment (Colin1860) [reply src/stack.rs:140]: Imagine a stack with 10000+ elements, the size-method does not scale and all just for adding numbers. Do you know what i mean?

File: src/stack.rs [hunk 3]
```diff
@@ -0,0 +1,285 @@
+//! A safe stack.
+//!
+//! The `Stack` allows inserting, removing and iterating it's elements.
+//!
+//! NOTE: This was written for a learning purpose.
+
```
Comment (Colin1860) [inline src/stack.rs:6]: If you'd be up for it, you could implement the from trait for the linked list: linkedlist -> stack so to speak

**Review by Colin1860 — CHANGES_REQUESTED** [review]

**Pull request merged**